packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -501,7 +512,10 @@ class Analytics implements IAnalytics {
       this.reset();
     }
 
-    this.userSessionManager?.setUserId(payload.userId);
+    // `null` value indicates that previous user ID needs to be retained
+    if (!isNull(payload.userId)) {
```

Comment:
why we are not able ti set null userId? what happens if user logs out? it should clear user id somehow correct?

Reply:
You're probably correct. But this is in line with our [v1.1 implementation](https://github.com/rudderlabs/rudder-sdk-js/blob/723955d6f408cf97ec7e1a17c4467abcdb2bff10/src/core/analytics.js#L708).
Setting `null` as the user ID would not clear the user. Instead, users make a `reset` call.

Reply:
Can we bring this into Ryan's attention and see if this is something we want to change in v3?